### PR TITLE
homePage updated

### PR DESCRIPTION
**Pull request by raffay0shahzad**
Kindly Review the PR for homePage

### CODE REVIEW

File: client/src/pages/CreateEventPage/index.jsx [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+                  htmlFor="eventImage"
+                  className="block text-sm font-medium text-gray-700"
+                >
+                  Event Image
+                </label>
+                <input
+                  type="file"
+                  name="eventImage"
+                  onChange={(event) => {
+                    setFieldValue("eventImage", event.currentTarget.files[0]);
```
Comment (ShanCoder110) [inline client/src/pages/CreateEventPage/index.jsx:183]: Move this function declaration above return just call it here

File: client/src/pages/CreateEventPage/index.jsx [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+              Create Event
+            </button>
+          </Form>
+        </div>
+      )}
+    </Formik>
+  );
+};
+
+EventForm.propTypes = {
```
Comment (ShanCoder110) [inline client/src/pages/CreateEventPage/index.jsx:248]: What is this doing?

File: client/src/pages/LoginPage/Form.jsx [hunk 3]
```diff
@@ -7,8 +7,7 @@ import * as yup from "yup";
 import { useNavigate } from "react-router-dom";
 import { useDispatch } from "react-redux";
 import { setLogin } from "../../state";
-import axios from 'axios';
-
+import axios from "axios";
 
 const registerSchema = yup.object().shape({
   name: yup.string().required("required"),
```
Comment (ShanCoder110) [inline client/src/pages/LoginPage/Form.jsx:13]: Make separate file

File: client/src/pages/LoginPage/Form.jsx [hunk 4]
```diff
@@ -39,67 +38,64 @@ const LoginForm = () => {
...
-      }
-    );
+  const register = async (values, onSubmitProps) => {
+    const formData = new FormData();
+    formData.append("jsonData", JSON.stringify(values));
+    console.log(formData);
+
+    try {
+      const response = await axios.post(
+        "http://localhost:3001/auth/register",
```
Comment (ShanCoder110) [inline client/src/pages/LoginPage/Form.jsx:48]: Same as above

File: client/src/pages/LoginPage/Form.jsx [hunk 5]
```diff
@@ -39,67 +38,64 @@ const LoginForm = () => {
...
+        values,
+        {
+          headers: {
```
Comment (ShanCoder110) [inline client/src/pages/LoginPage/Form.jsx:51]: Add headers in config file

File: client/src/pages/LoginPage/Form.jsx [hunk 6]
```diff
@@ -39,67 +38,64 @@ const LoginForm = () => {
...
-            showToast(loggedIn.message);
-          }
-          console.log("first");
-        } catch (error) {
-            showToast(error.response.data.message);
+  const login = async (values, onSubmitProps) => {
+    try {
+      const response = await axios.post(
+        "http://localhost:3001/auth/login",
+        values,
```
Comment (ShanCoder110) [inline client/src/pages/LoginPage/Form.jsx:74]: Move to constant and config also move base url to env

File: client/src/pages/LoginPage/Form.jsx [hunk 7]
```diff
@@ -121,85 +117,118 @@ const register = async (values, onSubmitProps) => {
...
+                  type="submit"
+                  className="flex w-full justify-center rounded-md bg-indigo-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
+                >
+                  {isLogin ? "LOGIN" : "REGISTER"}
+                </button>
+              </div>
+              <div className="mt-4 text-center">
+                <span
+                  onClick={() => {
+                    setPageType(isLogin ? "register" : "login");
```
Comment (ShanCoder110) [inline client/src/pages/LoginPage/Form.jsx:221]: Move function before return call it here

File: client/src/state/index.jsx [hunk 8]
```diff
@@ -1,27 +1,40 @@
...
-      setLogout: (state) => {
-        state.user = null;
-        state.token = null;
-      },
+  name: "auth",
+  initialState,
+  reducers: {
+    setLogin: (state, action) => {
+      state.user = action.payload.user;
+      state.token = action.payload.token;
```
Comment (ShanCoder110) [inline client/src/state/index.jsx:18]: Store token in local storage

**Review by ShanCoder110 — CHANGES_REQUESTED** [review]

File: client/src/pages/CreateEventPage/index.jsx (hunk 2 above)
Comment (raffay0shahzad) [reply client/src/pages/CreateEventPage/index.jsx:248]: It was giving some about not passing the props for Event form so looked it up on the internet

File: client/src/App.jsx [hunk 9]
```diff
@@ -2,28 +2,32 @@ import { BrowserRouter, Routes, Route, Navigate } from "react-router-dom";
 import { CssBaseline } from "@mui/material";
 import { useSelector } from "react-redux";
 
-import LoginPage from "./pages/LoginPage"
-import {HomePage} from "./pages/HomePage"
+import LoginPage from "./pages/LoginPage";
+import { HomePage } from "./pages/HomePage";
+import CreateEventPage from "./pages/CreateEventPage";
+//import Cart from "./pages/CartPage";
```
Comment (ShanCoder110) [inline client/src/App.jsx:8]: remove commented code

File: client/src/App.jsx [hunk 10]
```diff
@@ -2,28 +2,32 @@ import { BrowserRouter, Routes, Route, Navigate } from "react-router-dom";
...
-          </Routes>
+        <CssBaseline />
+        <Routes>
+          <Route path={LOGIN} element={<LoginPage />} />
+          <Route
+            path={HOME}
+            element={isAuth ? <HomePage /> : <Navigate to={LOGIN} />}
+          />
+          <Route path={CREATE_EVENT_PAGE} element={<CreateEventPage />} />
+          {/* <Route path={CART_ROUTE} element={<Cart />} /> */}
```
Comment (ShanCoder110) [inline client/src/App.jsx:26]: remove commented code

File: client/src/ValidationSchemas/index.js [hunk 11]
```diff
@@ -0,0 +1,28 @@
+import * as Yup from "yup";
+
+
```
Comment (ShanCoder110) [inline client/src/ValidationSchemas/index.js:3]: move schemas in different files

File: client/src/axiosConfig.js [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+
+const axiosInstance = axios.create({
+  baseURL: "http://localhost:3001",
+});
+
+export default axiosInstance;
+
+export const HEADERS = {
+    "Content-Type": "application/json",
+  };
```
Comment (ShanCoder110) [inline client/src/axiosConfig.js:11]: eol missing

File: client/src/ValidationSchemas/index.js [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+export const registerSchema = Yup.object().shape({
+    name: Yup.string().required("required"),
+    email: Yup.string().email("invalid email").required("required"),
+    password: Yup.string().required("required"),
+  });
+  
+  export const loginSchema = Yup.object().shape({
+    email: Yup.string().email("invalid email").required("required"),
+    password: Yup.string().required("required"),
```
Comment (ShanCoder110) [inline client/src/ValidationSchemas/index.js:27]: eol missing

File: client/src/components/Events.jsx [hunk 14]
```diff
@@ -0,0 +1,50 @@
+// src/components/Events.jsx
+import React, { useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+//import axios from "axios";
```
Comment (ShanCoder110) [inline client/src/components/Events.jsx:4]: remove commented code

File: client/src/pages/CreateEventPage/index.jsx [hunk 15]
```diff
@@ -0,0 +1,236 @@
+//import React from 'react';
```
Comment (ShanCoder110) [inline client/src/pages/CreateEventPage/index.jsx:1]: remove commented code

File: client/src/components/Events.jsx [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+
+const Events = () => {
+  const dispatch = useDispatch();
+  const [events, setEvents] = useState([]);
+  const [error, setError] = React.useState(null);
+
+  useEffect(() => {
+    const fetchEvents = async () => {
+      try {
+        const response = await axiosInstance.get(GET_EVENTS);
```
Comment (ShanCoder110) [inline client/src/components/Events.jsx:18]: for GET_EVENT be more specific like GET_EVENT_API_ROUTE or EVENT_API_ROUTE.getAllEvents

**Review by ShanCoder110 — CHANGES_REQUESTED** [review]
minor comments added

**Review by ShanCoder110 — APPROVED** [review]

**Pull request merged**